### PR TITLE
feat(proxy): List Entities

### PR DESCRIPTION
**Pull request by FintanH**
Allows us to list projects and users via the Peer and means that project and user retrieval is persistent across runs.

### CODE REVIEW

**Comment by FintanH** [comment]
Dependent on patch mentioned in https://github.com/radicle-dev/radicle-link/pull/190#discussion_r442916915

File: proxy/src/coco.rs [hunk 1]
```diff
@@ -83,7 +60,6 @@ impl Peer {
         let (api, _futures) = peer.accept()?;
         Ok(Self {
             api: Arc::new(Mutex::new(api)),
-            projects: Arc::new(Mutex::new(HashMap::new())),
```
Comment (xla) [inline proxy/src/coco.rs:86]: :heart_eyes_cat:

File: proxy/src/coco.rs [hunk 2]
```diff
@@ -101,15 +77,58 @@ impl Peer {
...
+    ///
+    /// # Errors
+    ///
+    /// The function will error if:
+    ///   * A lock was poisioned. See [`Self::with_api`].
+    #[allow(
+        clippy::wildcard_enum_match_arm,
+        clippy::match_wildcard_for_single_variants
+    )]
+    pub fn list_users(&self) -> Result<Vec<user::User<entity::Draft>>, error::Error> {
```
Comment (xla) [inline proxy/src/coco.rs:117]: Are we using this anywhere?

Comment (FintanH) [reply proxy/src/coco.rs:117]: Not yet, but I'm sure it'll be useful for feed-like features.

**Comment by FintanH** [comment]
This has a bug in it, given that I introduced it in `radicle-link` (see comment above). It might be worth waiting for https://github.com/radicle-dev/radicle-link/pull/197 to land, or else we merge and patch after. What do you think @xla?

File: proxy/src/coco.rs (hunk 2 above)
Comment (xla) [reply proxy/src/coco.rs:117]: Either we cover it with a test or we add when we need it, otherwise just feels like a liability.

**Comment by xla** [comment]
> This has a bug in it, given that I introduced it in `radicle-link` (see comment above). It might be worth waiting for [radicle-dev/radicle-link#197](https://github.com/radicle-dev/radicle-link/pull/197) to land, or else we merge and patch after. What do you think @xla?

Sound plan, wanna make this PR draft n the meantime?

**Comment by FintanH** [comment]
I don't think you can go back from a "Ready for Review" to "Draft". Maybe reject it and I'll fix it up. The radicle-link PR has been merged now.

**Pull request merged**